model/visor/processing.go

```diff
@@ -0,0 +1,201 @@
+
+	// ClaimedUntil marks the actor as claimed for processing until the set time
+	ClaimedUntil time.Time
+
+	// CompletedAt is the time the actor was read from the chain and its state read
+	CompletedAt time.Time
+
+	// ErrorsDetected contains any error encountered when reading the actor's state
+	ErrorsDetected string
+}
+
+func (p *ProcessingActor) PersistWithTx(ctx context.Context, tx *pg.Tx) error {
+	if _, err := tx.ModelContext(ctx, p).
+		OnConflict("do nothing").
+		Insert(); err != nil {
+		return fmt.Errorf("persisting processing actor: %w", err)
+	}
+	return nil
+}
+
+func (p *ProcessingActor) TipSetKey() (types.TipSetKey, error) {
+	return TipSetKeyFromString(p.TipSet)
+}
+
+func (p *ProcessingActor) ParentTipSetKey() (types.TipSetKey, error) {
+	return TipSetKeyFromString(p.ParentTipSet)
+}
+
+type ProcessingActorList []*ProcessingActor
+
+func (pl ProcessingActorList) PersistWithTx(ctx context.Context, tx *pg.Tx) error {
+	ctx, span := global.Tracer("").Start(ctx, "ProcessingActorList.PersistWithTx", trace.WithAttributes(label.Int("count", len(pl))))
+	defer span.End()
+	for _, p := range pl {
+		if err := p.PersistWithTx(ctx, tx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func NewProcessingMessage(ts *types.TipSet) *ProcessingMessage {
+	return &ProcessingMessage{
+		TipSet:  ts.Key().String(),
+		Height:  int64(ts.Height()),
+		AddedAt: time.Now(),
+	}
+}
+
+type ProcessingMessage struct {
+	tableName struct{} `pg:"visor_processing_messages"`
+
+	TipSet string `pg:",pk,notnull"`
+	Height int64  `pg:",use_zero"`
+
+	// AddedAt is the time the block was discovered and written to the table
+	AddedAt time.Time `pg:",notnull"`
+
+	// ClaimedUntil marks the block as claimed for message processing until the set time
+	ClaimedUntil time.Time
+
+	// CompletedAt is the time the block was read from the chain and its messages read
+	CompletedAt time.Time
+
+	// ErrorsDetected contains any error encountered when reading the block's messages
+	ErrorsDetected string
+}
+
+func (p *ProcessingMessage) PersistWithTx(ctx context.Context, tx *pg.Tx) error {
+	if _, err := tx.ModelContext(ctx, p).
+		OnConflict("do nothing").
+		Insert(); err != nil {
+		return fmt.Errorf("persisting processing actor: %w", err)
+	}
+	return nil
+}
+
+func (p *ProcessingMessage) TipSetKey() (types.TipSetKey, error) {
+	return TipSetKeyFromString(p.TipSet)
+}
+
+type ProcessingMessageList []*ProcessingMessage
+
+func (pl ProcessingMessageList) PersistWithTx(ctx context.Context, tx *pg.Tx) error {
+	ctx, span := global.Tracer("").Start(ctx, "ProcessingMessageList.PersistWithTx", trace.WithAttributes(label.Int("count", len(pl))))
+	defer span.End()
+	for _, p := range pl {
+		if err := p.PersistWithTx(ctx, tx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func TipSetKeyFromString(s string) (types.TipSetKey, error) {
+	if len(s) < 2 {
+		return types.EmptyTSK, xerrors.Errorf("invalid tipset")
+	}
+
+	s = s[1 : len(s)-1]
```

Comment:
Because we're using `.String()` to serialize into the database, which wraps with `{}`:
```
// String() returns a human-readable representation of the key.
func (k TipSetKey) String() string {
	b := strings.Builder{}
	b.WriteString("{")
	cids := k.Cids()
	for i, c := range cids {
		b.WriteString(c.String())
		if i < len(cids)-1 {
			b.WriteString(",")
		}
	}
	b.WriteString("}")
	return b.String()
}
```

I considered using JSON but it's possible that the Go JSON encoder can change over time. Raw bytes are also a possibility but less friendly to work with when manually querying the database.